index.js

```diff
@@ -771,6 +800,71 @@ SendStream.prototype.stream = function stream (path, options) {
 }
 
 /**
+ * Multipart Stream `path` to the response.
+ *
+ * @param {String} path
+ * @param {Object} options
+ * @api private
+ */
+
+SendStream.prototype.streamMultipart = function streamMultipart (path, options) {
+  var finished = false
+  var self = this
+  var res = this.res
+  var fileSize = options.fileSize
+  var BOUNDARY = options.boundary
+  var CRLF = '\r\n'
+  var stream
+
+  fs.open(path, 'r', function beginMultipartStream (err, fd) {
+    if (err) {
+      return self.onStatError(err)
+    }
+    // iterate through all the ranges
+    asyncSeries(options.ranges, function (range, idx, next) {
+      if (finished) return next()
+      var isLast = idx >= options.ranges.length - 1
+      var partHeaders = (
+        (idx ? CRLF : '') +
+        '--' + BOUNDARY + CRLF +
+        'Content-Type: ' + options.contentType + CRLF +
+        'Content-Range: ' + contentRange('bytes', fileSize, range) + CRLF +
+        CRLF
+      )
+      /* istanbul ignore next */
+      range.fd = isLegacyNodeVersion && idx ? null : fd
+      range.autoClose = isLast
+      stream = fs.createReadStream(path, range)
+      stream.on('error', next)
+      stream.on('end', function onpartend () { process.nextTick(next) })
+      if (!idx) self.emit('stream', stream)
```

Comment:
Not sure how you want to handle this. I'm creating a new read stream for every range/part so right now I'm only emitting the 'stream' event on the first iteration. I could emit this event on every iteration, but I feel like that might be confusing for a developer using this library since they would end up seeing this event get triggered N times, but the 'end' event would only get emitted once.